src/sql/parts/registeredServer/common/objectExplorerService.ts

```diff
@@ -65,16 +65,29 @@ export interface IObjectExplorerService {
 	onSelectionOrFocusChange: Event<void>;
 
 	getServerTreeView(): ServerTreeView;
+
+	getActiveConnections(): { connectionId: string, nodeInfo: sqlops.NodeInfo }[];
+
+	getChildren(connectionId: string, nodePath: string): Thenable<sqlops.NodeInfo[]>;
+
+	isExpanded(connectionId: string, nodePath: string): boolean;
+
+	expandNodeForConnection(connectionId: string, nodePath: string): Thenable<void>;
+
+	collapseNodeForConnection(connectionId: string, nodePath: string): Thenable<void>;
```

Comment:
A few thoughts on these methods:
- we've got 3 separate actions already. I'd guess we'll have more. Would a "getNode" action to return the Node class be better, and then hang actions off of this?
- I'd suggest a single method for expand/collapse such as `setExpandedState(TreeItemCollapsibleState.Collapsed)` rather than separate methods
- Similarly, do a `setSelected(true/false)` method since it's also useful to allow unselect or similar.
- Consider how to support multi-selection vs single-selection. VSCode has been moving towards supporting multi-select and we will no doubt need to also.